packages/renderer/src/lib/preferences/PreferencesConnectionCreationRendering.spec.ts

```diff
@@ -51,13 +55,28 @@ beforeAll(() => {
 });
 
 test('Expect that the create button is available', async () => {
+  const dispatch = createEventDispatcher();
+  const svelte = await import('svelte');
+  svelte.onMount = vi.fn().mockImplementation(async () => { 
+    svelte.onMount;
+    dispatch('pageLoaded');
+  });
   const callback = vi.fn();
-  await render(PreferencesConnectionCreationRendering, { properties, providerInfo, propertyScope, callback });
+  const {component} = await render(PreferencesConnectionCreationRendering, { properties, providerInfo, propertyScope, callback });
+  await waitOnMount(component);
   const createButton = screen.getByRole('button', { name: 'Create' });
   expect(createButton).toBeInTheDocument();
   expect(createButton).toBeEnabled();
 });
 
+function waitOnMount(component: PreferencesConnectionCreationRendering): Promise<any> {
+  return new Promise(resolve => {
+    component.$on('pageLoaded', function() {
+      resolve(component);
+    });
+  })  
+}
+
```

Comment:
probably you should replace the onMount and restore it in a beforeEach/afterEach call

you could replace the renderer method as well


with a custom `renderer` method doing
```
const originalOnMount = svelte.onMount;

beforeEach(() => {
svelte.onMount = {
try {
     originalOnMount();
     } finally {
     dispatch('pageLoaded');
     }
}
);
```

```
afterEach(() => {
svelte.onMount = originalOnMount

```

and for renderer
const originalRender = render;

```
beforeEach(() => {
render = {
  const result = originalRender(.....);
  return new Promise(resolve => {
    component.$on('pageLoaded', function() {
      resolve(result);
    });
  })  
```


in your code you're not handling error so you should add the dispatch inside a finally block of a try/finally

Reply:
Ops, i didn't notice i pushed my testing... my fault i was rebasing.
BTW this won't work or atleast i'm not able to make it work. You cannot use the dispatched there apparently.
It returns an error `Function called outside component initialization`

Reply:
I also tried with an eventEmitter, no luck

Reply:
Oh ok

Reply:
As i'm wasting too much time on it i propose three ways to progress
1) commenting the two existing tests and opening a PR to work on the onMount issue later
2) using the hack of having the dispatch on the svelte file
3) using the hack to export the variable and set it true when running the tests

Of course all three require to spend some time on the main problem but it's really blocking me and after 8 days i think it's better to move on.